pkg/sql/schema_changer.go

```diff
@@ -1771,3 +1767,51 @@ func init() {
 	}
 	jobs.RegisterConstructor(jobspb.TypeSchemaChange, createResumerFn)
 }
+
+// queueCleanupJobs checks for columns and indexes that have been dropped
+// and queues a job for the garbage collector to delete the data.
+func (sc *SchemaChanger) queueCleanupJobs(
+	ctx context.Context, scDesc *MutableTableDescriptor, txn *kv.Txn,
+	childJobs []*jobs.StartableJob,
+) ([]*jobs.StartableJob, error) {
+	// Create jobs for dropped columns / indexes to be deleted.
+	mutationID := scDesc.ClusterVersion.NextMutationID
+	span := scDesc.PrimaryIndexSpan()
+	var spanList []jobspb.ResumeSpanList
+	for i := len(scDesc.ClusterVersion.Mutations) + len(spanList); i < len(scDesc.Mutations); i++ {
+		spanList = append(spanList,
+			jobspb.ResumeSpanList{
+				ResumeSpans: []roachpb.Span{span},
+			},
+		)
+	}
+
+	// Only start a job if spanList has any spans. If len(spanList) == 0, then
+	// no mutations were enqueued by the primary key change.
+	if len(spanList) > 0 {
+		jobRecord := jobs.Record{
+			Description:   fmt.Sprintf("CLEANUP JOB for '%s'", sc.job.Payload().Description),
+			Username:      sc.job.Payload().Username,
+			DescriptorIDs: sqlbase.IDs{scDesc.GetID()},
+			Details: jobspb.SchemaChangeDetails{
+				TableID:        sc.tableID,
+				MutationID:     mutationID,
+				ResumeSpanList: spanList,
+				FormatVersion:  jobspb.JobResumerFormatVersion,
+			},
+			Progress:      jobspb.SchemaChangeProgress{},
+			NonCancelable: true,
+		}
+		job, err := sc.jobRegistry.CreateStartableJobWithTxn(ctx, jobRecord, txn, nil /* resultsCh */)
+		if err != nil {
+			return nil, err
+		}
+		log.VEventf(ctx, 2, "created job %d to drop previous indexes", *job.ID())
```

Comment:
this log is out of date for your case -- should it be pulled out of this method, or the message should be generalized